File: library/src/main/java/com/gmail/brymher/materialcalendar/State.java
package com.gmail.brymher.materialcalendar;

import org.threeten.bp.DayOfWeek;

public class State {
    public final CalendarMode calendarMode;
    public final DayOfWeek firstDayOfWeek;
    public final CalendarDay minDate;
    public final CalendarDay maxDate;
    public final boolean cacheCurrentPosition;
    public boolean showWeekDays = true;
    public int showOtherDates = MaterialCalendarView.SHOW_DEFAULTS;

    State(final StateBuilder builder) {
        calendarMode = builder.calendarMode;
        firstDayOfWeek = builder.firstDayOfWeek;
        minDate = builder.minDate;
        maxDate = builder.maxDate;
        cacheCurrentPosition = builder.cacheCurrentPosition;
        showWeekDays = builder.showWeekDays;
    }

    /**
     * Modify parameters from current state.
     */
    public StateBuilder edit() {
        return new StateBuilder(this);
    }
}

File: library/src/main/java/com/gmail/brymher/materialcalendar/State.kt
package com.gmail.brymher.materialcalendar

import android.annotation.SuppressLint
import org.threeten.bp.DayOfWeek
import org.threeten.bp.LocalDate
import org.threeten.bp.temporal.WeekFields
import java.util.*


class State(builder: StateBuilder) {


    val calendarMode: CalendarMode = builder.calendarMode
    val firstDayOfWeek: DayOfWeek = builder.firstDayOfWeek
    val minDate: CalendarDay? = builder.minDate
    val maxDate: CalendarDay? = builder.maxDate
    val cacheCurrentPosition: Boolean by lazy {
        builder.cacheCurrentPosition
    }
    val showWeekDays: Boolean = builder.showWeekDays

    /**
     * Modify parameters from current state.
     */
    fun edit(): StateBuilder {
        return StateBuilder(this)
    }

    companion object {
        @JvmStatic
        val CALENDAR_MODE = CalendarMode.MONTHS

        @SuppressLint("ConstantLocale")
        @JvmStatic
        val FIRST_DAY_WEEK = LocalDate.now().with(
            WeekFields.of(Locale.getDefault()).dayOfWeek(),
            1
        ).dayOfWeek
    }

}